File: rust/herb-linter/src/urls.rs
use std::sync::LazyLock;

pub const DOCS_BASE_URL: &str = "https://herb-tools.dev";

pub static DOCS_LINTER_BASE_URL: LazyLock<String> = LazyLock::new(|| format!("{DOCS_BASE_URL}/linter/rules"));

pub static RULE_CONFIGURATION_DOCUMENTATION_URL: LazyLock<String> =
  LazyLock::new(|| format!("{DOCS_BASE_URL}/configuration#setting-the-default-for-all-rules"));

pub fn rule_documentation_url(rule_id: &str) -> String {
  format!("{}/{rule_id}", *DOCS_LINTER_BASE_URL)
}

File: javascript/packages/linter/src/urls.ts
export const DOCS_BASE_URL = "https://herb-tools.dev"
export const DOCS_LINTER_BASE_URL = `${DOCS_BASE_URL}/linter/rules`
export const RULE_CONFIGURATION_DOCUMENTATION_URL = `${DOCS_BASE_URL}/configuration#setting-the-default-for-all-rules`

export function ruleDocumentationUrl(ruleId: string): string {
  return `${DOCS_LINTER_BASE_URL}/${ruleId}`
}
